src/agent/MessageSender.ts

```diff
@@ -55,6 +55,24 @@ export class MessageSender {
       connection: { id, verkey, theirKey },
     })
 
+    const session = this.transportService.findSession(connection.id)
+    if (
+      session?.inboundMessage?.hasReturnRouting() ||
+      session?.inboundMessage?.hasReturnRouting(outboundMessage.payload.threadId)
+    ) {
+      this.logger.debug(`Existing ${session.type} transport session has been found.`)
+      if (!session.keys) {
+        throw new AriesFrameworkError(`There are no keys for the given ${session.type} transport session.`)
+      }
```

Comment:
Do we want to throw an error here? Maybe just log that we couldn't find them and then move on to trying the outbound transports?

Reply:
Ok, I moved it inside the try-catch. What do you think?